docs/proposals/error-handling-guideline.md

```diff
@@ -0,0 +1,224 @@
+## Error output recommendation
+
+### Dos
+
+- Provide full description if the user input does not match what ORAS CLI expected. A full description should include the actual input received from the user and expected input
+- Use the capital letter ahead of each line of any error message
+- Print human readable error message. If the error message is mainly from the server and varies by different servers, tell users that the error response is from server. This implies that users may need to contact server side for troubleshooting.
+- Provide specific and actionable prompt message with argument suggestion or show the example usage for reference. (e.g, Instead of showing flag or argument options is missing, please provide available argument options and guide users to "--help" to view more examples)
+- If the actionable prompt message is too long to show in the CLI output, consider guide users to ORAS user guide or troubleshooting guide with the permanent link.
+- If the error message is not enough for troubleshooting, guide users to use "--verbose" to print much more detailed logs
+
+### Don'Ts
+
+- Do not use a formula-like or a programming expression in the error message. (e.g, `json: cannot unmarshal string into Go value of type map[string]map[string]string.`, or `Parameter 'xyz' must conform to the following pattern: '^[-\\w\\._\\(\\)]+$'`)
+- Do not use ambiguous expressions which mean nothing to users. (e.g, `Something unexpected happens`, or `Error: accepts 2 arg(s), received 0`)
+- Do not print irrelevant error message to make the output noisy. The more irrelevant output you produce, the longer it’s going to take the user to figure out what they did wrong.
+
+## How to write friendly error message
+
+### Recommended error message structure
+
+Here is a sample structure of an error message:
+
+```text
+Error:[Status code]: [Error description] 
+Usage: [Command usage]
+[Recommended solution]
+```
+
+- Status code is an optional information. If the error message is generated from the server side, it may include error code or [warn code](https://www.rfc-editor.org/rfc/rfc7234#section-5.5). It could be printed out alongside the error description.
+- Command usage is also an optional information but it's recommended to be printed out when user input doesn't follow the standard usage or examples. 
+- Recommended solution is required and should follow the general guiding principles described above.
+
+### Examples
+
+Here are some examples of writing error message with helpful prompt actionable information:
+
+#### Example 1: when no reference provided in `oras copy`
+
+Current behavior and output:
+
+```
+$ oras cp
+Error: accepts 2 arg(s), received 0
+```
+
+Suggested error message:
+
+```
+$ oras cp
+Error: "oras copy" requires exactly 2 arguments.
+Usage: oras copy [flags] <from>{:<tag>|@<digest>} <to>[:<tag>[,<tag>][...]]
+You need to specify two arguments as source and destination respectively. Run "oras copy -h" for more options and examples
```

Comment:
I don't find there is a general style guide that answers this question. Even though in Git, we also see it uses "you", "your" in its output messages. My suggestion is that ORAS should try to avoid using [personal pronouns](https://en.wikipedia.org/wiki/Personal_pronoun) but this is still allowed in some cases, such as questioning the CLI user.